lua/neogit/lib/buffer.lua

```diff
@@ -242,7 +289,7 @@ function Buffer:set_foldlevel(level)
 end
 
 function Buffer:replace_content_with(lines)
```

Comment:
How about making these *immediate* functions error if the buffer is not flushed?

That would be a implementation bug if not flushed, and will lead to text appearing despite the buffer being replaced. What do you think

Reply:
So something like this?

```lua
function Buffer:set_line(...)
  if #self.line_buffer > 0 then
    error()
  end

  ...
end
```